### PR TITLE
Template arbitrary data section

### PR DESCRIPTION
**Pull request by curlup**

### CODE REVIEW

File: cidc_schemas/schemas/participant.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
       "$comment": "With `in_doc_ref_pattern` here, there's no need to specify it each time cimac_participant_id is $ref'ed, constrain will be pulled in place by ref resolver automatically, assuring that it will be checked for every cimac_participant_id $ref.",
       "pattern": "^C[A-Z0-9]{3}[A-Z0-9]{3}$",
       "example": "CTTTP01",
-      "description": "Participant identifier assigned by the CIMAC-CIDC Network. Formated as: C??????."
+      "description": "Participant identifier assigned by the CIMAC-CIDC Network. Formated as: C?????? (first 7 characters of CIMAC ID)"
```
Comment (curlup) [inline cidc_schemas/schemas/participant.json:15]: unrelated

File: cidc_schemas/schemas/participant.json [hunk 2]
```diff
@@ -33,10 +33,6 @@
       "in_doc_ref_pattern": "/allowed_cohort_names/*",
       "$comment": "Values in this field are validated dynamically by `in_doc_ref_pattern`."
     },
-    "essential_patient_entry_number": {
```
Comment (curlup) [inline cidc_schemas/schemas/participant.json:36]: dropped because added checks started to detect collision on that field in the test data.

File: cidc_schemas/schemas/templates/manifests/pbmc_template.json [hunk 3]
```diff
@@ -75,8 +75,9 @@
                 "data_columns": {
                     "Path Concordance Verification": {
                         "entry (#)": {
-                            "merge_pointer": "2/essential_patient_entry_number",
-                            "type_ref": "participant.json#properties/essential_patient_entry_number"
+                            "do_not_merge": true,
```
Comment (curlup) [inline cidc_schemas/schemas/templates/manifests/pbmc_template.json:78]: dropped because added checks started to detect collision on that field in the test data

File: cidc_schemas/schemas/templates/manifests/plasma_template.json [hunk 4]
```diff
@@ -75,8 +75,9 @@
                 "data_columns": {
                     "Path Concordance Verification": {
                         "entry (#)": {
-                            "merge_pointer": "2/essential_patient_entry_number",
-                            "type_ref": "participant.json#properties/essential_patient_entry_number"
+                            "do_not_merge": true,
```
Comment (curlup) [inline cidc_schemas/schemas/templates/manifests/plasma_template.json:78]: dropped because added checks started to detect collision on that field in the test data

File: cidc_schemas/template.py [hunk 5]
```diff
@@ -525,18 +529,20 @@ def _load_keylookup(self) -> dict:
         """
 
         # create a key lookup dictionary
-        key_lu = {}
+        key_lu = defaultdict(dict)
```
Comment (curlup) [inline cidc_schemas/template.py:532]: key look up dict is now dict of dicts mapping through worksheet name (disambiguates similarly named template keys in different template tabs/worksheets)

File: cidc_schemas/template.py [hunk 6]
```diff
@@ -546,18 +552,51 @@ def _load_keylookup(self) -> dict:
...
+                    raw_val,
+                )
+            ],
+            [],
+        )
 
     def process_field_value(
-        self, key: str, raw_val, format_context: dict, eval_context: dict
+        self,
+        worksheet: str,
```
Comment (curlup) [inline cidc_schemas/template.py:595]: added worksheet to distinguish between similarly named fields in different template worksheets/tabs

File: cidc_schemas/schemas/participant.json (hunk 2 above)
Comment (jacoblurye) [reply cidc_schemas/schemas/participant.json:36]: Can you clarify why it's ok to drop this field, given that `additionalProperties` is still `false` for this schema? Do no participants in existing metadata have values for this field?

File: cidc_schemas/schemas/participant.json [hunk 7]
```diff
@@ -67,6 +63,11 @@
         "Other"
       ]
     },
+    "arbitrary_trial_specific_clinical_annotations":{
```
Comment (jacoblurye) [inline cidc_schemas/schemas/participant.json:66]: `additional_clinical_annotations`, maybe, to make this more concise?

File: cidc_schemas/template.py [hunk 8]
```diff
@@ -546,18 +552,51 @@ def _load_keylookup(self) -> dict:
...
+            [
+                AtomicChange(
+                    arbitrary_data_merge_pointer
+                    + "/"
+                    + self._sanitize_arbitrary_key(key),
+                    raw_val,
+                )
+            ],
+            [],
+        )
```
Comment (jacoblurye) [inline cidc_schemas/template.py:591]: Why not just return the `AtomicChange`?

**Review by jacoblurye — APPROVED** [review]
Nice!

File: cidc_schemas/schemas/participant.json (hunk 7 above)
Comment (curlup) [reply cidc_schemas/schemas/participant.json:66]: I don't think they are "additional" to anything

File: cidc_schemas/schemas/participant.json (hunk 2 above)
Comment (curlup) [reply cidc_schemas/schemas/participant.json:36]: right, I guess I won't be dropping the field then.
